apps/Bestiary_DB_Builder/Program.cs

```diff
@@ -0,0 +1,219 @@
+            Debug.WriteLine("Done extracting");
+            //delete zip
+            File.Delete(zipfile);
+            Debug.WriteLine("Done deleting zipfile");
+
+            string cwd = Directory.GetCurrentDirectory();
+            //navigate to bestiary folder
+            var intermediate_path = Path.Join(cwd, "5etools");
+            //navigate again because the stupid folder is named something like 5etools-mirror-3-5etools-src-fe58f72
+            //this sohuld only return 1 directory
+            var dirs = Directory.GetDirectories(intermediate_path);
+            var bestiary_path = dirs[0];
+            bestiary_path = Path.Join(bestiary_path, "data", "bestiary");
+            Debug.WriteLine("Besteiary path is " + bestiary_path);
+            var bestiary_files = Directory.GetFiles(bestiary_path);
+
+            List<StopGapMonsterClass> monsters = new List<StopGapMonsterClass>();
+            //ignore everything but the bestiary files (eg the fluff files) for now
+
+            foreach (var bestiary_file in bestiary_files)
+            {
+                var fname = Path.GetFileName(bestiary_file);
+                if (!fname.StartsWith("bestiary"))
+                {
+                    Debug.WriteLine("Skipping file: " + bestiary_file);
+                    continue;
+                }
+                Debug.WriteLine("Processing file: " + bestiary_file);
+                //read file
+                //TODO: Use streamreader if files are super big but surely this doesn't happen
+                Monster_root mon_root = JsonConvert.DeserializeObject<Monster_root>(File.ReadAllText(bestiary_file));
+                monsters.AddRange(mon_root.Monsters);
+
+            }
+
+            return monsters;
+
+        }
+
+        public static async Task<List<StopGapMonsterClass>> Download_Data_Src5etools()
+        {
+            using var playwright = await Playwright.CreateAsync();
+            var browser = await playwright.Chromium.LaunchAsync(new BrowserTypeLaunchOptions
+            {
+                Headless = true,
+
+            });
+            var page = await browser.NewPageAsync(new BrowserNewPageOptions
+            {
+                UserAgent = @"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
+            });
+
+            var response = await page.GotoAsync("https://5e.tools/data/bestiary/index.json");
+            string re = await response.TextAsync();
+            //Debug.WriteLine(re);
+            JObject jobject = JObject.Parse(re);
+
+            var urls_to_download = new List<string>();
+
+            foreach (JToken jtoken in jobject.PropertyValues())
+            {
+                var bestiary_file = jtoken.ToString();
+                const string base_url = "https://5e.tools/data/bestiary/";
+                var file_url = base_url + bestiary_file;
+                urls_to_download.Add(file_url);
+
+            }

```

Comment:
nit: Try to use Linq more to reduce imperative code.

```suggestion
            var urls_to_download = jobject.PropertyValues()
               .Select(jtoken => {
                    var bestiary_file = jtoken.ToString();
                    const string base_url = "https://5e.tools/data/bestiary/";
                    var file_url = base_url + bestiary_file;
                    return file_url;
               }).ToList()
```

Reply:
nit: Move the 'const' expression to the top of the file since it doesn't change.